docs/branching.md

```diff
@@ -0,0 +1,98 @@
+- TODO – Automate contract deployments in **development**
+
+Deploying new contracts in the release branch:
+- :stop_sign: - _NOPE_
+- We cannot deploy any `elixir-omg` code that is incompatible with the currently deployed contracts in **staging** and **production**
+
+### 2 - Production deployment, validating a new version for network upgrade
+
+Branches and environments:
+- `master` is automatically deployed to **development** environment
+- `master` is automatically deployed to **staging-v0-2** environment
+  - This assumes that during the process of validating a network upgrade, all work merged onto `master` will get deployed to for the upgrade.
+- `v0.1` is automatically deployed to **staging-v0-1** environment
+  - Keep this environment around for hotfixes
+- `v0.1` is manually deployed to **production-v0-1** environment
+
+Deploying new contracts in `master`:
+- Same as Scenario 1, with **development** and **staging-v0-2** environments
+
+Deploying new contracts in `v0.1`:
+- :stop_sign: _NOPE_
+
+### 3 - Production deployment, ready to deploy network upgrade to production, ongoing development
+
+When we're confident of the stability on **staging-v0-2** and ready to go to Private Alpha, create the `v0.2` branch from `master` for both `elixir-omg` and `plasma-contracts` repos.
+
+Most importantly, we're confident about the contracts. A contract redeployment in this phase would have the most impact.
+
+This is a _feature freeze_ for the new version (`v0.2` branch). Try to minimize merging changes from `master` to any of the release branches.
+
+Branches and environments:
+- `master` is automatically deployed to **development** environment
+- `v0.2` is automatically deployed to **staging-v0-2** environment
+  - changes to this release branch will be merged into `master`, as needed
+- `v0.2` is manually deployed to **production-v0-2** environment
+- `v0.1` is automatically deployed to **staging-v0-1** environment
+  - changes to the release branch will be merged into `master`, as needed
+- `v0.1` is manually deployed to **production-v0-1** environment
+
+Deploying new contracts to `master`:
+- Same as Scenario 1
+
+Deploying new contracts to `v0.1`
+- :stop_sign: _NOPE_
+
+Deploying new contracts to `v0.2`
+- Manually deploy to **staging-v0-2** and **production-v0-2** environments, _if absolutely necessary_
+
+### 4 - Two production deployments, ongoing development
+
+We will have two production environments during the network upgrade, so that users have the opportunity to exit the old environment and deposit into the new environment.
+
+Everything the same as Scenario 3 except - Deploying new contracts to `v0.2`
+- :stop_sign: _NOPE_
+
+Once this phase ends, we take down the older `production-v0-1` and `staging-v0-1` and return to Scenario 1. We may want to consider continuing to run a watcher for an old version a longer period of time.
```

Comment:
curious, what's the reason to keep watcher when operator is stopping already? (To see the exit games for 2 weeks?_

Reply:
> curious, what's the reason to keep watcher when operator is stopping already? (To see the exit games for 2 weeks?_

 This too I think, but not only.

The chain is stopped but there still might be invalid exits we'd like to have an eye on. Also this is what people will do - "their" watchers will continue to do their job connected to the defunct child chain, until all tokens are exited. So we're mimicking other people's situation here.